src/components/chart/useInitChart.ts

```diff
@@ -0,0 +1,96 @@
+import { useLayoutEffect, useRef } from "react";
+import { createChart } from "lightweight-charts";
+import { ChartApiRef, ChartOptions } from "./types";
+import { chartDefaultOptions } from "./chartDefaultOptions";
+
+export const useInitChart = ({
+  container,
+  ...rest
+}: {
+  container: HTMLElement;
+} & ChartOptions) => {
+  const { onClick, onCrosshairMove, ...restOptions } = rest;
+
+  const chartApiRef = useRef<ChartApiRef>({
+    _chart: null,
+    api() {
+      if (this._chart === null) {
+        this._chart = createChart(container, {
+          ...chartDefaultOptions,
+          ...restOptions,
+        });
+
+        if (onClick) {
+          this._chart.subscribeClick(onClick);
+        }
+
+        if (onCrosshairMove) {
+          this._chart.subscribeCrosshairMove(onCrosshairMove);
+        }
+
+        this._chart.timeScale().fitContent();
+      }
+
+      return this._chart;
+    },
+    update(options: ChartOptions) {
+      if (this._chart === null) {
+        return;
+      }
+
+      const { onClick: nextOnClick, onCrosshairMove: nextOnCrosshairMove, ...restOptions } = options;
+
+      if (onClick !== nextOnClick) {
```

Comment:
onClick here would be always the same on the start of init this context

Reply:
if you want to check it in this type you should save link to function in this object too